ietf/secr/sreq/views.py

```diff
@@ -105,18 +105,32 @@ def get_lock_message(meeting=None):
         meeting = get_meeting(days=14)
     return meeting.session_request_lock_message
 
-def get_requester_text(person,group):
-    '''
-    This function takes a Person object and a Group object and returns the text to use in the
-    session request notification email, ie. Joe Smith, a Chair of the ancp working group
-    '''
-    roles = group.role_set.filter(name__in=('chair','secr'),person=person)
+
+def get_requester_text(person, group):
+    """
+    This function takes a Person object and a Group object and returns the text to use
+    in the session request notification email, ie. Joe Smith, a Chair of the ancp
+    working group
+    """
+    roles = group.role_set.filter(name__in=("chair", "secr", "ad"), person=person)
     if roles:
-        return '%s, a %s of the %s working group' % (person.ascii, roles[0].name, group.acronym)
-    if group.parent and group.parent.role_set.filter(name='ad',person=person):
-        return '%s, a %s Area Director' % (person.ascii, group.parent.acronym.upper())
-    if person.role_set.filter(name='secr',group__acronym='secretariat'):
-        return '%s, on behalf of the %s working group' % (person.ascii, group.acronym)
+        rolename = str(roles[0].name)
+        return "%s, a%s %s of the %s %s%s" % (
+            person.name,
+            "n" if rolename[0].lower() in "aeiou" else "",
+            rolename,
+            group.acronym,
+            group.type,
+            " group" if str(group.type)[-1].lower() != "g" else "",
```

Comment:
This doesn't do what is requested at all.
It produces this:
```
>>> [f'{slug}{" group" if slug[-1].lower() != "g" else ""}' for slug in GroupTypeName.objects.values_list('slug',flat=True)]
['adhoc group', 'ag', 'area group', 'dir group', 'review group', 'edappr group', 'edwg', 'iab group', 'iabasg', 'iana group', 'iesg', 'ietf group', 'individ group', 'irtf group', 'ise group', 'isoc group', 'nomcom group', 'program group', 'rag', 'rg', 'sdo group', 'team group', 'wg', 'adm group', 'rfcedtyp group']
```
What we want is output like ['iab program', 'iab administrative support group', 'review team',]

This should use GroupTypeName.verbose_name, but we probably need some data cleaning first:
```
>>> for typename in GroupTypeName.objects.all():
...     print(f"{typename.slug}: {typename.verbose_name}")
... 
adhoc: Ad Hoc Group Type
ag: Area Group
area: Area
dir: Area Directorate
review: 
edappr: 
edwg: 
iab: Internet Architecture Board
iabasg: IAB Administrative Support Group
iana: Internet Assigned Numbers Authority
iesg: Internet Engineering Steering Group
ietf: Internet Engineering Task Force
individ: An Individual
irtf: Internet Research Task Force
ise: Independent Stream Editor
isoc: The Internet Society
nomcom: IETF/IAB Nominating Committee
program: 
rag: Research Area Group
rg: Research Group
sdo: Standards Organization
team: Team
wg: Working Group
adm: Administrative Group
rfcedtyp: RFC Editor Group
```